### ISSUE TITLE
Packet: 422 You must specify a valid facility parameter

### ISSUE THREAD

**Issue by majst01** [kubicorn/kubicorn#588]
Hi,

im following the packet instructions from here:
http://kubicorn.io/documentation/packet-walkthrough.html
I didnt modify the created _state/nameofproject/cluster.yaml 

But 
```
kubicorn apply -v 10 kubernetes-poc
```

ends with:
```
2018-03-17T15:21:44+01:00 [✖]  Error during apply of atomic reconciler, attempting clawback: POST https://api.packet.net/projects/<my_project_id>/devices: 422 You must specify a valid facility parameter
```
looking at the very beginning of the verbose log output already shows an empty Location:

```json
2018-03-17T15:21:43+01:00 [▶]  device.Apply devices actual 0, expected 1
2018-03-17T15:21:43+01:00 [▶]  creating server kubernetes-poc.master-0: packngo.DeviceCreateRequest{HostName:"kubernetes-poc.master-0", Plan:"baremetal_1", Facility:"", OS:"ubuntu_16_04", BillingCycle:"hourly", ProjectID:"fba6ca1f-0f69-4d7b-902d-1a9ad9dc10e4", UserData:"#!/usr/bin/env bash

2018-03-17T15:21:35+01:00 [ℹ]  Query existing resources
2018-03-17T15:21:35+01:00 [▶]  PacketModel: {"0":{"Identifier":"","name":"kubicorn-kubernetes-poc","Tags":[]},"1":{"Identifier":"","name":"kubernetes-poc","Tags":null,"User":"","PublicKeyFingerprint":"","PublicKeyData":"","PublicKeyPath":""},"2":{"Identifier":"","name":"kubernetes-poc.master","Location":"","Type":"","OS":"","SSHFingerprint":"","BootstrapScripts":null,"Count":0,"ServerPool":{"metadata":{"creationTimestamp":null},"minCount":1,"maxCount":1,"type":"master","name":"kubernetes-poc.master","image":"ubuntu_16_04","size":"baremetal_1","bootstrapScripts":["bootstrap/packet_k8s_ubuntu_16.04_master.sh"]},"Tags":null,"ProjectID":""},"3":{"Identifier":"","name":"kubernetes-poc.node","Location":"","Type":"","OS":"","SSHFingerprint":"","BootstrapScripts":null,"Count":0,"ServerPool":{"metadata":{"creationTimestamp":null},"minCount":1,"maxCount":1,"type":"node","name":"kubernetes-poc.node","image":"ubuntu_16_04","size":"baremetal_2","bootstrapScripts":["bootstrap/packet_k8s_ubuntu_16.04_node.sh"]},"Tags":null,"ProjectID":""}}
```

Any hints to dig into this are welcome, i try to figure out where this error starts, but up to now i didnt find the right place in the source.

**Comment by majst01** [comment]
Update:

```diff

diff --git a/cloud/packet/public/resources/device.go b/cloud/packet/public/resources/device.go
index a4be232d..4186edb7 100644
--- a/cloud/packet/public/resources/device.go
+++ b/cloud/packet/public/resources/device.go
@@ -194,6 +194,7 @@ func (r *Device) Apply(actual, expected cloud.Resource, immutable *cluster.Clust
                        ProjectID:    projectID,
                        BillingCycle: "hourly",
                }
+               createRequest.Facility = "ewr1"
                logger.Debug("creating server %s: %v", hostname, createRequest)
                device, _, err = Sdk.Client.Devices.Create(createRequest)
                if err != nil {
```

With this it works as expected, so somewhere the configuration from cluster.yaml --> providerConfig --> location gets lost. Digging further.

**Comment by xmudrii** [comment]
Fixed by #589
